### ISSUE TITLE
why we need fake_A_seg_list.append(empty) 

### ISSUE THREAD

**Issue by TJJTJJTJJ** [sangwoomo/instagan#7]
```python
fake_A_seg_list = self.fake_A_seg_list + [self.fake_A_seg_sng]  # not detach
for i in range(self.ins_iter - idx - 1):
	fake_A_seg_list.append(empty)

self.fake_A_seg_mul = torch.cat(fake_A_seg_list, dim=1)
self.fake_A_mul = torch.cat([self.fake_A_img_sng, self.fake_A_seg_mul], dim=1)
```
Hi, your code is perfect. But I have some questions. 
According your code, I think that self.fake_A_mul is to be GAN\_loss in the global in your paper, is this right?
More than, what's the point of "fake_A_seg_list.append(empty)", because in "NLayerSetDiscriminator", empty will be abandoned, so it confuses me, please help me to understand the code?
And, in your paper, you use pix2pix model to predict the segmentation, how can I reproduce this or when you release this? If it can be done in just input the image, mask can be predict by the model, transfer can also be done by the model, it's unbelievable!
Thank you, sir!

**Comment by sangwoomo** [comment]
1. You are exactly right! sng = local, mul = global in our code/paper. Sorry for the confusion!
2. **Edited:** DataParallel module in PyTorch requires samples (to be parallelized) to have identical size. Hence, I appended empty masks to match the size.
3. I didn't released pix2pix model since it is not our contribution. But you can easily reproduce it using `pix2pix_model.py` implemented by [the authors of CycleGAN](https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix), which is also included in our code.
Specifically, put your images and masks in trainA/trainB respectively, and simply run
```
python train.py --dataroot ./datasets/your_dataset --name your_dataset_pix2pix --model pix2pix
```

**Comment by Ko8e-24** [comment]
Hi, I want to transfer a single photo from google, since there is no mask for that photo, I need to train the pix2pix model  with the MHP dataset first. Then I should use the pix2pix model to produce a mask for that photo. Finally, I put both that photo and its mask into the pants2skirt_mhp model. Am I right?
